sdk/core/core-client-rest/test/sendRequest.spec.ts

```diff
@@ -27,15 +27,90 @@ describe("sendRequest", () => {
       await sendRequest("POST", mockBaseUrl, mockPipeline, { body: foo });
     });
 
-    it("should handle request body as string", async () => {
+    it("should handle request body as non-json string without content type", async () => {
       const mockPipeline: Pipeline = createEmptyPipeline();
-      const expectedBody = "foo";
+      const expectedBody = '"foo"';
```

Comment:
I'm not sure this is the expected string we want. I think we should keep "foo" instead of '"foo"'. It looks like we are running a non-JSON string through `JSON.stringify(nonJSON)` which results in the extra quotes which will cause issues since it is a different string. We should not run a non-json string through JSON.stringify

Reply:
I think this is because we have default the logic to application json in content type, if it's not set https://github.com/Azure/azure-sdk-for-js/pull/27343/files#diff-c7a7b43a730d4a3de597de20c6de199f8468c0d29e27ccc52b6b9a26a5afbd52R77-R84 

Previously, there's a bug in the core about this part, I think, we will default the content type to application json if it's not specified, but we will not json.stringify the request body at the same time.

Reply:
Can we add a check in get contentType so that if the body is string and no content-Type is set to set the content-type as plain-text to avoid the extra quotes? Having the extra quotes results in an invalid request to be sent

Reply:
I think I am okay with that approach.

Reply:
Updated it as we set the default content-type to text/plain if it's non-json string in the body, but remain to be application/json if it's json string. let me know if you have any more concerns.